src/node_main.cc

```diff
@@ -83,26 +86,12 @@ int wmain(int argc, wchar_t *wargv[]) {
 extern char** environ;
 #endif  // __linux__
 
-namespace node {
-  extern bool linux_at_secure;
-}  // namespace node
-
 int main(int argc, char *argv[]) {
-#if defined(__linux__)
-  char** envp = environ;
-  while (*envp++ != nullptr) {}
-  Elf_auxv_t* auxv = reinterpret_cast<Elf_auxv_t*>(envp);
-  for (; auxv->a_type != AT_NULL; auxv++) {
-    if (auxv->a_type == AT_SECURE) {
-      node::linux_at_secure = auxv->a_un.a_val;
-      break;
-    }
-  }
-#endif
```

Comment:
This can't be moved, it breaks AT_SECURE detection.  Libuv modifies argv and environ so the AT_SECURE check needs to happen very early.

Reply:
Ugh, I was afraid of that. Thanks for the clarification on that.